cmd/juju/caas/add.go

```diff
@@ -195,19 +225,34 @@ func (c *AddCAASCommand) Run(ctxt *cmd.Context) error {
 		return errors.Errorf("CAData attribute should be a string")
 	}
 
-	newCloud := cloud.Cloud{
+	newCloud := jujucloud.Cloud{
 		Name:           c.caasName,
 		Type:           c.caasType,
 		Endpoint:       currentCloud.Endpoint,
-		AuthTypes:      []cloud.AuthType{credential.AuthType()},
+		AuthTypes:      []jujucloud.AuthType{credential.AuthType()},
 		CACertificates: []string{cloudCAData},
 	}
 
+	cloudRegion := c.cloudRegion
+	if cloudRegion == "" {
+		cloudRegion, err = c.getClusterRegion(ctx, newCloud, credential)
```

Comment:
We can even store there error here to print out later if required. ie it's done here as debug but if the jaas controller needs it and we couldn't get it, we can print the actual error